tutorials/general/t8_tutorial_build_cmesh.cxx

```diff
@@ -0,0 +1,532 @@
+  t8_cmesh_set_join (cmesh, 4, 5, 4, 2, 0);
+
+  /*
+   * Definition of the first tree
+   */
+  /* Tetrahedron 1 vertices */
+  vertices[0] = 0.43;
+  vertices[1] = 0;
+  vertices[2] = 2;
+
+  vertices[3] = 0;
+  vertices[4] = 0;
+  vertices[5] = 1;
+
+  vertices[6] = 0.86;
+  vertices[7] = -0.5;
+  vertices[8] = 1;
+
+  vertices[9] = 0.86;
+  vertices[10] = 0.5;
+  vertices[11] = 1;
+
+  /* Classification of the vertices for the first tree */
+  t8_cmesh_set_tree_vertices (cmesh, 0, vertices, 4);
+
+  /*
+   * Definition of the second tree
+   */
+  /* Tetrahedron 2 vertices */
+  for (i = 0; i < 3; i++) {
+
+    vertices[i] = vertices[i] + (i == 0 ? 1 + 0.86 : 0);
+    vertices[3 + i] = vertices[6 + i] + (i == 0 ? 1 : 0);
+    vertices[9 + i] = vertices[9 + i] + (i == 0 ? 1 : 0);
+  }
+
+
+  vertices[6] = 1 + 2 * 0.86;
+  vertices[7] = 0;
+  vertices[8] = 1;
+
+  /* Classification of the vertices for the second tree */
+  t8_cmesh_set_tree_vertices (cmesh, 1, vertices, 4);
+
+  /*
+   * Definition of the third tree
+   */
+  /* Prism 1 vertices */
+  vertices[0] = 0;
+  vertices[1] = 0;
+  vertices[2] = 0;
+
+  vertices[3] = 0.86;
+  vertices[4] = -0.5;
+  vertices[5] = 0;
+
+  vertices[6] = 0.86;
+  vertices[7] = 0.5;
+  vertices[8] = 0;
+
+  /* Translate +1 in z-axis for the upper vertices */
+  for (i = 0; i < 3; i++) {
+    vertices[9 + 3 * i] = vertices[3 * i];
+    vertices[9 + 3 * i + 1] = vertices[3 * i + 1];
+    vertices[9 + 3 * i + 2] = vertices[3 * i + 2] + 1;
+  }
+
+  /* Classification of the vertices for the third tree */
+  t8_cmesh_set_tree_vertices (cmesh, 2, vertices, 6);
+
+  /*
+   * Definition of the fourth tree
+   */
+  /* Prism 2 vertices */
+  for (i = 0; i < 3; i++) {
+    vertices[3 + i] = vertices[i] + (i == 0 ? 1 + 2 * 0.86 : 0);
+    vertices[6 + i] = vertices[6 + i] + (i == 0 ? 1 : 0);
+  }
+
+  vertices[0] = 0.86 + 1;
+  vertices[1] = -0.5;
+  vertices[2] = 0;
+
+  /* Translate +1 in z-axis for the upper vertices */
+  for (i = 0; i < 3; i++) {
+    vertices[9 + 3 * i] = vertices[3 * i];
+    vertices[9 + 3 * i + 1] = vertices[3 * i + 1];
+    vertices[9 + 3 * i + 2] = vertices[3 * i + 2] + 1;
+  }
+
+  /* Classification of the vertices for the fourth tree */
+  t8_cmesh_set_tree_vertices (cmesh, 3, vertices, 6);
+
+
+  /*
+   * Definition of the fifth tree
+   */
+  /* Pyramid vertices */
+
+  vertices[0] = 0.86;
+  vertices[1] = -0.5;
+  vertices[2] = 0;
+
+  vertices[3] = 1.86;
+  vertices[4] = -0.5;
+  vertices[5] = 0;
+
+  vertices[6] = 0.86;
+  vertices[7] = 0.5;
+  vertices[8] = 0;
+
+  vertices[9] = 1.86;
+  vertices[10] = 0.5;
+  vertices[11] = 0;
+
+  vertices[12] = 1.36;
+  vertices[13] = 0;
+  vertices[14] = -0.5;
+
+  /* Classification of the vertices for the fifth tree */
+  t8_cmesh_set_tree_vertices (cmesh, 4, vertices, 5);
+
+  /*
+   * Definition of the sixth tree
+   */
+  /* Hex coordinates */
+  /* Translate +1 in z-axis for the upper vertices */
+  for (i = 0; i < 4; i++) {

```

Comment:
```suggestion
  for (i = 0; i < 4; i++) {
    vertices[3 * i + 2] = vertices[3 * i + 2] - 1;
```

Reply:
Set the "lower" base vertices of hex (vertices with id 0, 1, 2, 3) equal to base vertices of the pyramid translated by -1 in the z-axis.